packages/cardpay-sdk/sdk/utils/conversions.ts

```diff
@@ -80,12 +80,16 @@ export async function gasPriceInToken(provider: JsonRpcProvider, tokenAddress: s
   if (tokenAddress === wrappedNativeToken) {
     return gasPriceInNativeTokenInWei;
   }
+
   let rate = await tokenPairRate(provider, tokenAddress, wrappedNativeToken);
-  let rateAdjusted = adjustRate(rate);
-  // Convert the current gas price which is in native token to the token we want to pay the gas with.
-  return gasPriceInNativeTokenInWei
-    .mul(new BN(rateAdjusted.numerator.toString()))
-    .div(new BN(rateAdjusted.denominator.toString()));
+  let rateFraction = new Fraction(rate.numerator, rate.denominator).multiply(rate.scalar);
+  let rateBN = new BN(rateFraction.numerator.toString()).div(new BN(rateFraction.denominator.toString()));
```

Comment:
why don't we use rateAdjust function here?

Reply:
`adjustRate` returned the same rate as the one passed in. I'm not sure it's working correctly

Reply:
What is the value of `rateBN` here for CTSC case? This is adjustRate's result `1994.82`.

Reply:
```
rateBN.toString()
'1994'
```

Testing adjustRate function:

```
rate.toFixed()
=> '1994.8200'

rateAdjusted = adjustRate(rate);
new BN(rateAdjusted.numerator.toString()).div(new BN(rateAdjusted.denominator.toString())).toString()
=> '1994'
```

Reply:
Hmm, I think we can use the rate rather than rateAdjusted or rateBN.

Reply:
How do you mean? To use the fixed decimal value?

Reply:
Like this:

```
BigNumber.from(rate.toFixed(0))
    .mul(gasPriceInNativeTokenInWei)
    .mul(new BN(rate.scalar.denominator.toString()))
    .div(new BN(rate.scalar.numerator.toString()));
```

Reply:
Doesn't expressing the rate as `BigNumber.from(rate.toFixed(0))` reduce precision of the calculation? I opted into using a pure BigNumber solution to keep the advantages of calculating the whole procedure with BigNumber

Reply:
I think there is no different between `BigNumber.from(rate.toFixed(0))` and `new BN(rateFraction.numerator.toString()).div(new BN(rateFraction.denominator.toString()));`. Yeah but it is not necessary for now, we can do it in another PR and maybe we will use FixedNumber instead of BigNumber.